crates/nu-command/src/strings/str_/escape_glob.rs

```diff
@@ -0,0 +1,92 @@
+use nu_cmd_base::input_handler::{operate, CmdArgument};
+use nu_engine::CallExt;
+use nu_protocol::ast::Call;
+use nu_protocol::ast::CellPath;
+use nu_protocol::engine::{Command, EngineState, Stack};
+use nu_protocol::Category;
+use nu_protocol::{Example, PipelineData, ShellError, Signature, Span, SyntaxShape, Type, Value};
+
+#[derive(Clone)]
+pub struct SubCommand;
+
+struct Arguments {
+    cell_paths: Option<Vec<CellPath>>,
+}
+
+impl CmdArgument for Arguments {
+    fn take_cell_paths(&mut self) -> Option<Vec<CellPath>> {
+        self.cell_paths.take()
+    }
+}
+
+impl Command for SubCommand {
+    fn name(&self) -> &str {
+        "str escape-glob"
+    }
+
+    fn signature(&self) -> Signature {
+        Signature::build("str escape-glob")
+            .input_output_types(vec![
+                (Type::String, Type::String),
+                (
+                    Type::List(Box::new(Type::String)),
+                    Type::List(Box::new(Type::String)),
+                ),
+                (Type::Table(vec![]), Type::Table(vec![])),
+                (Type::Record(vec![]), Type::Record(vec![])),
+            ])
+            .allow_variants_without_examples(true)
+            .rest(
+                "rest",
+                SyntaxShape::CellPath,
+                "For a data structure input, turn strings at the given cell paths into substrings.",
+            )
+            .category(Category::Strings)
+    }
+
+    fn usage(&self) -> &str {
+        "Escape glob pattern."
+    }
+
+    fn search_terms(&self) -> Vec<&str> {
+        vec!["pattern", "list", "ls"]
+    }
+
+    fn run(
+        &self,
+        engine_state: &EngineState,
+        stack: &mut Stack,
+        call: &Call,
+        input: PipelineData,
+    ) -> Result<PipelineData, ShellError> {
+        let cell_paths: Vec<CellPath> = call.rest(engine_state, stack, 0)?;
+        let cell_paths = (!cell_paths.is_empty()).then_some(cell_paths);
+        let args = Arguments { cell_paths };
+        operate(action, args, input, call.head, engine_state.ctrlc.clone())
+    }
+
+    fn examples(&self) -> Vec<Example> {
+        vec![Example {
+            description: "escape glob pattern before list",
+            example: r#"let f = test[a]; ls ($f | str escape-glob)"#,
```

Comment:
I think this is invalid without quotes.
```suggestion
            example: r#"let f = 'test[a]'; ls ($f | str escape-glob)"#,
```